lib/evm/opcodes.ts

```diff
@@ -170,7 +170,11 @@ let codes: any = {
 }
 
 const istanbulOpcodes: any = {
+  0x31: ['BALANCE', 700, true],
+  0x3f: ['EXTCODEHASH', 700, true],
   0x46: ['CHAINID', 2, false],
+  0x47: ['SELFBALANCE', 5, false],
+  0x54: ['SLOAD', 800, true],
```

Comment:
Hmm, this breaks with the concept of having gas price changes collected in the Common library (e.g. for [SLOAD](https://github.com/ethereumjs/ethereumjs-common/blob/master/src/hardforks/tangerineWhistle.json#L10)), I only expected the Istanbul opcode list to include the newly introduced opcodes (`SELFBALANCE`).

For the moment this would introduce a lot of inconsistency - we also have added other gas price changes for Istanbul in the [istanbul.json](https://github.com/ethereumjs/ethereumjs-common/blob/master/src/hardforks/istanbul.json) file.

Do you have a bigger plan for a complete transition here respectively a strong reasoning for this break apart? Otherwise I would very much suggest we add the price changes to the Common library.

Reply:
Hmm, I didn't know the base cost of opcodes are also updated in Common (in addition to extra gas parameters like `ecAdd`). If that has been the intention, there's no corresponding code in the VM to read those base gas costs from Common:

https://github.com/ethereumjs/ethereumjs-vm/blob/edffcde9696283799efc4ab17d0c3e481618c727/lib/evm/interpreter.ts#L105-L112

If we want to keep base gas costs in Common we'd have to:

- Make sure costs for all opcodes are in `Common` (not the case now, [searching](https://github.com/ethereumjs/ethereumjs-common/search?q=swap&unscoped_q=swap) `swap` in ethereumjs-common returned empty)
- Remove them from `opcodes.ts` to only have one source of truth
- Modify `Interpreter` to read costs from Common

Reply:
Hmm, maybe we are talking side by side a bit? What is with `SLOAD` e.g. - the EIP is stating that this is changing gas costs from 200 to 800 - and the 200 price is just "normally" defined in the tangerine whistle json file in the Common library (see link above).

This is just a "normal" gas price change, isn't it?

Reply:
Ah, just had another look. See the point, these gas costs are not taken by Common at all atm. Maybe this is just more of some inconsistency/incompleteness in the Common library.

Anyhow, you are right. Will approve here.